src/averageAges.js

```diff
@@ -54,11 +76,30 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  let peopleWithMothers = people.filter(person => person.mother);
+
+  onlyWithSon && (
+    peopleWithMothers = peopleWithMothers.filter(person => person.sex === 'm')
+  );
+
+  const { totalAgeDiff, validCount } = peopleWithMothers
+    .reduce(
+      (acc, person) => ({
+        totalAgeDiff: acc.totalAgeDiff + (people
+          .find(mother => mother.name === person.mother) ? person.born - people
```

Comment:
`.find(mother => mother.name === person.mother)`

you are doing it 3 times inside the reduce callback
save it into a variable inside the callback